src/smt/sygus_solver.h

```diff
@@ -37,13 +38,27 @@ class SmtSolver;
  * A solver for sygus queries.
  *
  * This class is responsible for responding to check-synth commands. It calls
- * check satisfiability using an underlying SmtSolver object.
+ * check satisfiability using a separate SolverEngine "subsolver".
  *
- * It also maintains a reference to a preprocessor for implementing
- * checkSynthSolution.
+ * This solver operates in two modes.
+ *
+ * If in incremental mode, then the "main" SolverEngine for SyGuS inputs only
+ * maintains a (user-context) dependent state of SyGuS assertions, as well as
+ * assertions corresponding to (recursive) function definitions. The subsolver
+ * that solves SyGuS conjectures may be called to checkSat multiple times,
```

Comment:
I'm a bit confused by this. The goal of multiple check-sats would be to allow getting multiple solutions? If so clarify this, because currently it's not clear what is the motivation to have this since any change in the conjecture triggers a destruction of the solver.